utils/parse.js

```diff
@@ -116,6 +116,11 @@ exports.default = function parse(path, content, context) {
   parserOptions = Object.assign({}, parserOptions);
   parserOptions.ecmaFeatures = Object.assign({}, parserOptions.ecmaFeatures);
 
+  // fallback to the ecma version in languageOptions when using flat config
+  if (context.languageOptions) {
+    parserOptions.ecmaVersion = parserOptions.ecmaVersion || context.languageOptions.ecmaVersion;
```

Comment:
I would suggest removing this change in favor of what I added to this PR: https://github.com/import-js/eslint-plugin-import/pull/3061 which gets both `ecmaVersion` and `sourceType`.  Also changes to `utils` need to be released separate.  So it may make more sense to let that other PR go through and absorb that.

Reply:
Yeah this'll go away once #3061 is landed - I'm not removing it for now because I need it for CI to be green and Jordan wants to cherry-pick stuff out anyway so there will be at least one "grand rebase" of this pull request before it gets merged 😄

Reply:
Looks like #3061 is ready to merge. I think the collective JS world is going to let out a sigh of relief!
Thank you so, so  much for all the effort @G-Rath!